common/oatbox/extension/script/ScriptAction.php

```diff
@@ -51,180 +77,274 @@
      * Invoke
      * 
      * This method makes the script invokable programatically.
-     * 
+     *
+     * @param array $params
+     *
+     * @throws \common_exception_Error
+     *
      * @return \common_report_Report
      */
     public function __invoke($params)
     {
-        $this->optionsDescription = $this->provideOptions();
-        $beginScript = microtime(true);
-        
-        // Display help?
-        if ($this->displayUsage($params)) {
-            return $this->usage();
-        }
-        
-        // Build option container.
+        $report = new Report(
+            Report::TYPE_INFO,
+            $this->provideDescription() . "\n"
+        );
+
         try {
+            // Collecting possible options.
+            $this->optionsDescription = array_merge(
+                $this->provideTraitOptions(),
+                $this->provideOptions()
+            );
+
+            // Build option container.
             $this->options = new OptionContainer(
-                $this->optionsDescription, 
+                $this->optionsDescription,
                 $params
             );
-        } catch (\Exception $e) {
-            return new Report(
-                Report::TYPE_ERROR,
-                $e->getMessage()
-            );
-        }
 
-        // Run the userland script.
-        $report = $this->run();
+            // Initializes the trait options.
+            $report = $this->initializeTraitOptions($report);
+
+            // Run the userland script.
+            $report->add($this->run());
```

Comment:
As explained on slack, the *implementer MUST be in control of the generation of the initial report*. The main reason is that calling scripts might want to rely on the data returned via the report. With your change, the [following script](https://github.com/oat-sa/extension-tao-operations/blob/master/scripts/tools/CreateClusterBackofficeUsers.php#L171) will not work anymore because it is waiting for some data from the root report of another script.

I appreciate the try, but please go back to the initial pull request where the modifications I made make the legacy stuff able to continue working.